Management/Clients/AirQualityDataClient.cs

```diff
@@ -74,22 +74,22 @@ public AirQualityDataClient(IConfiguration configuration)
         /// <param name="countryCode">country of interest eg. US.</param>
         /// <param name="cancellationToken">used to signal that the asynchronous task should cancel itself.</param>
         /// <returns>A <see cref="Task"/> representing the asynchronous operation, with a status code.</returns>
-        public async Task<Dictionary<City, int>> CalculateScoresAsync(
-            IEnumerable<City> cities,
+        public async Task<(State, double)> CalculateScoreForStateAsync(
             State state,
             CountryCode countryCode,
             CancellationToken cancellationToken)
         {
-            var cityResult = new List<(City, int)>();
+            var sampledCities = await GetDefaultCitiesAsync(state, countryCode, cancellationToken);
 
-            foreach (var city in cities)
+            var cityBag = new ConcurrentBag<(City, int)>();
+            var cityTasks = sampledCities.Select(async city =>
             {
-                cityResult.Add(await GetSingleCityAsync(city, state, countryCode, cancellationToken));
-            }
+                var result = await GetSingleCityAsync(city, state, countryCode, cancellationToken);
```

Comment:
Other client not necessary need to implement get city score and do similar thing like I did.
The only reason I did this is because the API I used does not support return state data.
If the api you are using can return state data directly, go for it.